OutOfSchool/OutOfSchool.BusinessLogic/Services/CompetitiveEventService.cs

```diff
@@ -86,20 +107,93 @@ public async Task Delete(Guid id)
     {
         logger.LogTrace($"Deleting CompetitiveEvent with Id = {id} started.");
 
-        var entity = new CompetitiveEvent() { Id = id };
+        var entity = await competitiveEventRepository.GetById(id);
 
         try
         {
             await competitiveEventRepository.Delete(entity).ConfigureAwait(false);
 
             logger.LogTrace($"CompetitiveEvent with Id = {id} succesfully deleted.");
         }
-        catch (DbUpdateConcurrencyException)
+        catch (Exception) // DbUpdateConcurrencyException
         {
             logger.LogError("Deleting failed. CompetitiveEvent with Id = {Id} doesn't exist in the system", id);
             throw new ArgumentOutOfRangeException(
                 nameof(id),
                 localizer[$"CompetitiveEvent with Id = {id} doesn't exist in the system"]);
         }
     }
+
+    private async Task ChangeJudges(CompetitiveEvent currentCompetitiveEvent, List<JudgeDto> judgeDtoList)
+    {
+        var deletedIds = currentCompetitiveEvent.Judges
```

Comment:
It looks like `.Where` will be better as we will have entity to delete instead of Id

Reply:
Thanks for the suggestion! I plan to improve the code by working with entities instead of just IDs, which will make the logic more efficient and avoid unnecessary steps.